03_launch_mgmt_cluster.sh

```diff
@@ -10,19 +10,148 @@ source lib/releases.sh
 # shellcheck disable=SC1091
 source lib/network.sh
 
-if [ "${EPHEMERAL_CLUSTER}" == "minikube" ]; then
-  IRONIC_HOST="${CLUSTER_URL_HOST}"
-  export IRONIC_HOST_IP="${CLUSTER_PROVISIONING_IP}"
+export IRONIC_HOST="${CLUSTER_URL_HOST}"
+export IRONIC_HOST_IP="${CLUSTER_PROVISIONING_IP}"
+
+sudo mkdir -p "${IRONIC_DATA_DIR}"
+sudo chown -R "${USER}:${USER}" "${IRONIC_DATA_DIR}"
+
+# Create certificates and related files for TLS
+if [ "${IRONIC_TLS_SETUP}" == "true" ]; then
+  export IRONIC_BASE_URL="https://${CLUSTER_URL_HOST}"
+
+  export IRONIC_CACERT_FILE="${IRONIC_CACERT_FILE:-"${WORKING_DIR}/certs/ironic-ca.pem"}"
+  export IRONIC_CAKEY_FILE="${IRONIC_CAKEY_FILE:-"${WORKING_DIR}/certs/ironic-ca.key"}"
+  export IRONIC_CERT_FILE="${IRONIC_CERT_FILE:-"${WORKING_DIR}/certs/ironic.crt"}"
+  export IRONIC_KEY_FILE="${IRONIC_KEY_FILE:-"${WORKING_DIR}/certs/ironic.key"}"
+
+  export IRONIC_INSPECTOR_CACERT_FILE="${IRONIC_INSPECTOR_CACERT_FILE:-"${WORKING_DIR}/certs/ironic-ca.pem"}"
+  export IRONIC_INSPECTOR_CAKEY_FILE="${IRONIC_INSPECTOR_CAKEY_FILE:-"${WORKING_DIR}/certs/ironic-ca.key"}"
+  export IRONIC_INSPECTOR_CERT_FILE="${IRONIC_INSPECTOR_CERT_FILE:-"${WORKING_DIR}/certs/ironic-inspector.crt"}"
+  export IRONIC_INSPECTOR_KEY_FILE="${IRONIC_INSPECTOR_KEY_FILE:-"${WORKING_DIR}/certs/ironic-inspector.key"}"
+
+  pushd "${WORKING_DIR}"
+  mkdir -p "${WORKING_DIR}/certs"
+  pushd "${WORKING_DIR}/certs"
+
+  # Generate CA Key files
+  if [ ! -f "${IRONIC_CAKEY_FILE}" ]; then
+    openssl genrsa -out "${IRONIC_CAKEY_FILE}" 2048
+  fi
+  if [ ! -f "${IRONIC_INSPECTOR_CAKEY_FILE}" ]; then
+    openssl genrsa -out "${IRONIC_INSPECTOR_CAKEY_FILE}" 2048
+  fi
+
+  # Generate CA cert files
+  if [ ! -f "${IRONIC_CACERT_FILE}" ]; then
+    openssl req -x509 -new -nodes -key "${IRONIC_CAKEY_FILE}" -sha256 -days 1825 -out "${IRONIC_CACERT_FILE}" -subj /CN="ironic CA"/
+  fi
+  if [ ! -f "${IRONIC_INSPECTOR_CACERT_FILE}" ]; then
+    openssl req -x509 -new -nodes -key "${IRONIC_INSPECTOR_CAKEY_FILE}" -sha256 -days 1825 -out "${IRONIC_INSPECTOR_CACERT_FILE}" -subj /CN="ironic inspector CA"/
```

Comment:
By default, we have set the same CA for Ironic and Ironic-inspector, but wouldn't this generate a new CA for ironic inspector?

Reply:
yes, if the file given is different than the ironic one. By default it is the ironic CA, that was already generated, so the file exists, so it is skipped